src/proj.rs

```diff
@@ -124,15 +124,18 @@ impl Area {
 }
 
 /// Easily get a String from the external library
-pub(crate) fn _string(raw_ptr: *const c_char) -> Result<String, ProjError> {
-    let c_str = unsafe { CStr::from_ptr(raw_ptr) };
+pub(crate) unsafe fn _string(raw_ptr: *const c_char) -> Result<String, ProjError> {
+    assert!(!raw_ptr.is_null());
+    let c_str = CStr::from_ptr(raw_ptr);
     Ok(str::from_utf8(c_str.to_bytes())?.to_string())
 }
 
 /// Look up an error message using the error code
 fn error_message(code: c_int) -> Result<String, ProjError> {
-    let rv = unsafe { proj_errno_string(code) };
-    _string(rv)
+    unsafe {
+        let rv = proj_errno_string(code);
+        _string(rv)
```

Comment:
Continuing my previous comment about `_string` safety– because we use `proj_errno_string` and `_string` _in conjunction_ here, this function is safe. No matter what value of `code` we pass to our `error_message` function, we'll never reference invalid memory (under the assumption `proj_errno_string` _always_ returns a valid string pointer) so we don't need to mark this function as `unsafe`

Reply:
That makes sense! Would you preserve this rationale in a comment?

Reply:
> under the assumption `proj_errno_string` always returns a valid string pointer

Lo and behold, `proj_errno_string` does _not_ always return a valid string pointer:

https://github.com/OSGeo/PROJ/blob/5e077729274f5d28e137e1a41f7d3350146614ef/src/strerrno.cpp#L12-L14

https://github.com/OSGeo/PROJ/blob/5e077729274f5d28e137e1a41f7d3350146614ef/src/strerrno.cpp#L37-L75

Specifically, it returns a null pointer if `code == 0`, so we should handle that here by adding a `if rv.is_null()` check. I opened a new GitHub Issue for that since I feel this pull request has already exceeded the initial scope I intended: https://github.com/georust/proj/issues/73. So prior to this pull request, if one called `error_message(0)`, we would dereference a null pointer in `_string`. After this pull request, we'll panic because of the `assert` I added in `_string`.